### ISSUE TITLE
Quality degradation

### ISSUE THREAD

**Issue by shnhrrsn** [kean/Nuke#499]
I'm noticing image quality seems to get worse on every app launch for images that have already been downloaded.

Is it possible Nuke is re-encoding each time? With enough app opens I end up with pretty bad jpeg artifacts.

If I delete all cache and relaunch, quality goes back to normal.

I'm using Nuke v10.3.1

This is what my pipeline looks like:

```swift
private let pipeline = ImagePipeline {
  $0.isProgressiveDecodingEnabled = false
  $0.isDecompressionEnabled = true
  $0.callbackQueue = dispatchQueue

  let config = ImagePipeline.Configuration.withDataCache
  $0.dataLoader = config.dataLoader
  $0.dataCache = config.dataCache
  $0.dataCachePolicy = .storeAll
  ($0.dataCache as? DataCache)?.sizeLimit = 1024 * 1024 * 300 // 300 MB
}
```

And then this is what my request looks like:

```swift
pipeline.imagePublisher(with: ImageRequest(
  urlRequest: urlRequest, 
  processors:[ImageProcessors.Resize(size: targetSize)]
))
```

**Comment by kean** [comment]
It's a regression introduced in Nuke 10.0. The [fix](https://github.com/kean/Nuke/pull/500) is available in [Nuke 10.3.2](https://github.com/kean/Nuke/releases/tag/10.3.3).

**Comment by shnhrrsn** [comment]
Thanks for the quick fix 🙏
